core/src/main/java/org/infinispan/topology/ClusterCacheStatus.java

```diff
@@ -172,6 +184,21 @@ private boolean addMember(Address joiner, CacheJoinInfo joinInfo) {
             return false;
          }
 
+         // Validate if the member is allowed to join
+         if (persistentState.isPresent()) {
```

Comment:
I have added another test to both Dist and Repl tests which reverses the order in which the cachemanagers are initialized. This means that the view order will be reversed on restart and the coordinator will be different, just to make things a bit more realistic.
The persisted state is identical between all nodes in an existing cluster (just the timestamps will differ) so it doesn't matter who is coordinator and who is a humble member. The conditions I want to avoid are nodes without state joining "stateful" clusters before the initial formation, and nodes with state joining stateless clusters AT ANY TIME (as that might corrupt their local state).
If all conditions are met, and the initial cluster has formed, new (stateless) nodes may join the cluster at will.